### DISCUSSION TITLE
How to set an "Owned Attribute" to a new Note created from a script

### DISCUSSION THREAD

**Discussion by PhilB-RLB** [TriliumNext/Trilium#2352] [Q&A] [answered]
Hello,

I would like to know if it's possible, when creating a New Note from a script, to set the value of an owned attribute.

**Comment by zadam** [comment]
Yes, e.g. like this:

```javascript
const note = api.createTextNote(parentNoteId, 'Note title', 'content').note;
note.setLabel("myLabel", "value");
```

**Reply by PhilB-RLB** [reply]
Hello Zadam,

thanks for you help.
Unfortunately it didn't seem to work.

My note is created by the following command :
const newNoteObject = api.createTextNote(todayNote.noteId, "Note Title", "Note Text");

Then I have tried to set the value of the label "Titre_Livre" by using :
newNoteObject.setLabel("Titre_Livre", "Book Title");
but the Owned Attribute "Titre_Livre" is not set for the new Note.

Could you tell me what is wrong with my code ?

**Reply by zadam** [reply]
FTFY:

```
const newNoteObject = api.createTextNote(todayNote.noteId, "Note Title", "Note Text").note;
```

See the `.note` on the end. The method returns an object and you need to pick the note object from that.

As a tip, use `console.log()` to figure out what is in variables (and watch console).

**Reply by PhilB-RLB** [reply]
Ok, thank you very much. It's work now !!!

Please, delete or close my previous post (https://github.com/zadam/trilium/discussions/2337) as I've found how to use attributes in order to populate the fields of my new created note.

**Reply by PhilB-RLB** [reply]
I've another question about the way to get the value of an attribute.

For a single value attribute, it's ok to get its value by :
const VALUE = Note.getLabelValue("Label_Name");

Now, if we use multi-value attribute, how to get all the values affected to the same label ?
With the previous js code, if we have the following attribute for a note : #Label_Name=Value1 #Label_Name=Value2,
we only get Value1 in the VALUE variable.

**Reply by zadam** [reply]
```javascript
for (const label of note.getLabels('label_name')) {
    console.log(label.value);
}
```

**Reply by PhilB-RLB** [reply]
Thank you again but could you explain in which variable is stored the attribute values (sorry for my poor JavaScript skills).
If I use you code :
for (const label of note.getLabels('label_name')) {
    console.log(label.value);
}
what is the variable I have to use in order to , for example, assign these values to the text of a new created note :
NOTE_TEXT = <wich variable ?>
const newNoteObject = api.createTextNote(todayNote.noteId, Note_Title, NOTE_TEXT).note;

**Reply by zadam** [reply]
I probably don't understand your question. "value" of an attribute is in the "value" attribute, as in the example: `label.value`.

**Reply by PhilB-RLB** [reply]
Ok, so if I want tu use the values of my multi-value attributes in the text of my note I use :
const newNoteObject = api.createTextNote(todayNote.noteId, Note_Title, label.value).note;

**Reply by PhilB-RLB** [reply]
To be more precise, if I have the following multi-value attributes
#my_attribute="Text01" #my_attribute="Text02"
I want to obtain the following text in my note : Text01 Text02

**Reply by PhilB-RLB** [reply]
Ok Zadam, I've found how to proceed in order to get all the value of my multi-value attributes.

let VAR_Auteur = "";
            for (label of Form_Note.getLabels('Auteur_e')) {
                // console.log(label.value);
                VAR_Auteur += label.value
                VAR_Auteur += "&nbsp-&nbsp"
            }

Thank you for you help.

**Comment by PhilB-RLB** [comment, marked as answer]
Answered
